File: WebServer/websites/beamor/models/map/MapInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WebServer.websites.beamor.models.map
{
    public class MapInfo
    {
        public string MapName { get; set; }
        public Tileset[] Tilesets { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
    }
}

File: GameClient/classes/model/MapInfo.ts
class MapInfo implements IMapInfo
{
    public MapName: string;
    public Tilesets: Tileset[];
    public Width: number;
    public Height: number;

    public static GetMapInfo(mapInfo: IMapInfo): MapInfo
    {
        var buffer = new MapInfo();
        buffer.MapName = mapInfo.MapName;
        buffer.Tilesets = Tileset.GetTilesets(mapInfo.Tilesets);
        buffer.Width = mapInfo.Width;
        buffer.Height = mapInfo.Height;

        return buffer;
    }
}